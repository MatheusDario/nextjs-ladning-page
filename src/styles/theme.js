export const theme = {
  colors: {
    white: '#f0f0f2',
    blue: '#023059',
    lightblue: '#065473',
    aquablue: '#08748c',
    green: '#02735e',
    lightgreen: '#6ab04c',
    lightgray: ' #cecece',
    red: '#dc143c',
    black: '#000000',
  },
  spacing: {
    xmall: '.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
  },
  font: {
    sizes: {
      xmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
    },
    family: {
      default: "'Lato', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
  breakpoints: {
    mobile: '@media (max-width:768px)',
    tests: '(max-width:768px)',
  },
};
