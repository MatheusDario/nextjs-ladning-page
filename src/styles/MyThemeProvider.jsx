import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import React from "react";
import { GlobalStyles } from "./global-styles";

export default function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}

// MyThemeProvider.propTypes = {
//   children: P.node.isRequired,
// };
