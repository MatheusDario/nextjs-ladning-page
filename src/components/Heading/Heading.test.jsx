/* eslint-disable no-unused-vars */
import { screen } from '@testing-library/react';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.blue,
      'font-size': theme.font.sizes.huge,
      'text-transform': 'uppercase',
    });
  });

  it('should match a snapshot', () => {
    const { container } = renderTheme(<Heading>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render white color', () => {
    renderTheme(<Heading $colordark={false}>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render small heading size', () => {
    renderTheme(<Heading size="small">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.small,
    });
  });

  it('should render medium heading size', () => {
    renderTheme(<Heading size="medium">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.medium,
    });
  });

  it('should render big heading size', () => {
    renderTheme(<Heading size="big">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.xlarge,
    });
  });

  it('should render huge heading size', () => {
    renderTheme(<Heading size="huge">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.huge,
    });
  });

  it('should correct font-size when using mobile', () => {
    renderTheme(
      <Heading as="h2" size="huge">
        Texto
      </Heading>,
    );
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(screen.getByRole('heading', { name: 'Texto' })).toHaveStyleRule(
      'font-size',
      theme.font.sizes.xlarge,
      {
        media: theme.breakpoints.tests,
      },
    );
  });

  it('should render uppercase letters', () => {
    renderTheme(<Heading $uppercase>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render lowercase letters', () => {
    renderTheme(<Heading $uppercase={false}>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'text-transform': 'none',
    });
  });

  it('should render correct elemente', () => {
    const { container } = renderTheme(<Heading as="h4">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });
    const h6 = container.querySelector('h4');

    expect(h6.tagName.toLowerCase()).toBe('h4');
  });
});
