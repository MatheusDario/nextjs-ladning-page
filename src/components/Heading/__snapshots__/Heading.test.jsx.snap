// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match a snapshot 1`] = `
.c0 {
  color: #023059;
  font-size: 5.6rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  Texto
</h1>
`;
